moveit_commander/src/moveit_commander/planning_scene_interface.py

```diff
@@ -380,3 +381,38 @@ def make_plane(self, name, pose, normal=(0, 0, 1), offset=0):
         co.planes = [p]
         co.plane_poses = [pose.pose]
         return co
+
+
+def acm_set_default(acm, obj, allow):
+    """Set default collision behavior for obj"""
+    if obj not in acm.default_entry_names:
+        acm.default_entry_names.append(obj)
+        acm.default_entry_values.append(allow)
+    else:
+        idx = acm.default_entry_names.index(obj)
+        acm.default_entry_values[idx] = allow
+
+
+def acm_set_allowed(acm, obj, other=None, allow=True):
+    """Allow collisions between obj and other"""
+    if other is None:
+        acm.set_default(obj, allow)
+        return
+
+    other_idx = acm.entry_names.index(other)
+    if obj not in acm.entry_names:
+        acm.entry_names.append(obj)
+        for entry in acm.entry_values:
+            entry.enabled.append(allow)
+        acm.entry_values.append(
+            AllowedCollisionEntry(enabled=[allow for i in range(len(acm.entry_names))])
+        )
+        acm.entry_values[-1].enabled[other_idx] = allow
+    else:
+        obj_idx = acm.entry_names.index(obj)
+        acm.entry_values[obj_idx].enabled[other_idx] = allow
+        acm.entry_values[other_idx].enabled[obj_idx] = allow
+
+
+AllowedCollisionMatrix.set_default = acm_set_default
+AllowedCollisionMatrix.set_allowed = acm_set_allowed
```

Comment:
Can you please comment on how usual/reasonable you think these monkey patches are?
I've never seen anyone patch ROS message objects before (granted, I didn't look for it) and at first glance it looks useful, but I also think it's a can of worms I'd prefer closed...

Reply:
This simply augments the standard ROS-message class `AllowedCollisionMatrix` with two new functions.

Reply:
I know what it does. I'm not a fan of monkey patching, but of course the resulting interface looks nice when used.
I don't know how roundabout it would be to add it directly in moveit_msgs instead of here for example.

Reply:
I am not aware of a possibility to add methods directly in moveit_msgs. Of course, this would be even nicer. If you know an option to do so, please implement that in a follow-up PR.